client.go

```diff
@@ -966,15 +966,44 @@ func (f *File) readChunkAt(ch chan result, b []byte, off int64) (n int, err erro
 	return
 }
 
+func (f *File) readAtSequential(b []byte, off int64) (int, error) {
+	readed := 0
+
+	for {
+		endRead := int64(math.Min(float64(readed+f.c.maxPacket), float64(len(b))))
```

Comment:
Converting these values through `float64` just so we can use `math.Min` is not a good idea. I mean, loss of accuracy is pretty unlikely (at 53 bits of accuracy, that’s 8 pebibytes after all. But really, it’s the significant speed loss in converting between floating point and integers.

Rather, we can just start our own buffer starting at our index point, then test the length, and truncate it, if it is too long:
```
rb := b[read:]
if len(rb) > f.c.maxPacket) {
  rb = rb[:f.c.maxPacket]
}
```

Reply:
better, thank you!